python/tests/test_nessie_cli.py

```diff
@@ -314,8 +315,20 @@ def test_merge() -> None:
 
     # Passing detached commit-id plus a _different_ hash-on-ref --> error
     execute_cli_command(["merge", f"dev@{dev_hash}", "-c", main_hash, "-o", main_hash], ret_val=1)
+    merge_output = execute_cli_command(["merge", "dev", "-c", main_hash])
 
-    execute_cli_command(["merge", "dev", "-c", main_hash])
+    branches = ReferenceSchema().loads(execute_cli_command(["--json", "branch"]), many=True)
+    refs = {i.name: i.hash_ for i in branches}
+
+    assert_that(merge_output).is_equal_to(
+        "The following 1 commits were merged onto main:"
+        '\n{} "test message" \nResultant hash on main after merge: {}\n'.format(refs["dev"], refs["main"])
+    )
+
+    # if we try to merge again from dev to main we get an error
+    # this is because there is now a conflict as the same commit exists on both branches
+    merge_error = execute_cli_command(["merge", "dev", "-c", refs["main"]], ret_val=1)
+    assert "Client Error Conflict: The following keys have been changed in conflict:" in merge_error
     logs = simplejson.loads(execute_cli_command(["--json", "log"]))
```

Comment:
nit: verifying the logs is still part of the happy path test... so should come before triggering the conflict error imo

Reply:
but possible we want to keep it to verify that not only do we get an error also nothing was written server-side, so keeping as is is fine... maybe add newline inbetween